### PR TITLE
KAADO-49: Add Sharing function in frontend

### PR DESCRIPTION
**Pull request by nasudon**
- Add Sharing button component
- Update card page and card details page

### CODE REVIEW

File: src/components/ShareButton.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+import { ShareIcon } from '@/components/icons/ShareIcon';
+import { ISharing } from '@/models/Sharing';
+
+export const ShareButton = ({
+    getSharingData,
+}: {
+    getSharingData: () => Promise<{ sharing?: ISharing; error?: any }>;
+}) => {
+    const [shareMsg, setshareMsg] = useState('');
```
Comment (tsoibet) [inline src/components/ShareButton.tsx]: `setshareMsg` -> `setShareMsg`

File: src/components/ShareButton.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    };
+
+    return (
+        <>
+            <button onClick={handleShareClick}>
+                <ShareIcon />
+            </button>
+            {shareMsg && (
+                <div
+                    className={`fixed transition-all w-max transform left-1/2 -translate-x-1/2 border px-2 py-3 ${
```
Comment (tsoibet) [inline src/components/ShareButton.tsx]: Please set background colour.

**Review by tsoibet — CHANGES_REQUESTED** [review]

File: src/components/ShareButton.tsx (hunk 1 above)
Comment (nasudon) [reply src/components/ShareButton.tsx]: Fixed: https://github.com/tsoibet/kaado/pull/24/commits/e2be4b0b35a92cb7ee93ea2015163c35f9c77dc9

File: src/components/ShareButton.tsx (hunk 2 above)
Comment (nasudon) [reply src/components/ShareButton.tsx]: Fixed: https://github.com/tsoibet/kaado/pull/24/commits/e2be4b0b35a92cb7ee93ea2015163c35f9c77dc9
I also update the style of the toast box

**Review by tsoibet — APPROVED** [review]
LGTM

**Pull request merged**